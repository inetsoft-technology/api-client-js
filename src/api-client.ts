/*
 * Copyright 2019 InetSoft Technology
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export * from "./client/http-adapter";
export * from "./client/database/database-api";
export * from "./client/datasource/data-source-api";
export * from "./client/file/file-api";
export * from "./client/printer/printer-api";
export * from "./client/report/report-api";
export * from "./client/schedule/schedule-api";
export * from "./client/server/server-api";
export * from "./client/sheet/sheet-api";
export * from "./client/sheet/viewsheet/viewsheet-api";
export * from "./client/sheet/worksheet/worksheet-api";
export * from "./client/client";
