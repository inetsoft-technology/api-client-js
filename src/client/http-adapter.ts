/*
 * Copyright 2019 InetSoft Technology
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Interface for classes that provide an adapter to an HTTP client implementation.
 */
export interface HttpAdapter<R> {
   /**
    * Executes an HTTP GET request.
    *
    * @param url     the URL.
    * @param params  the request query parameters, if any.
    * @param headers the HTTP request headers, if any.
    *
    * @return the HTTP response body.
    */
   get(url: string, params?: {[name: string]: string},
       headers?: {[name: string]: string}): Promise<R>;

   /**
    * Executes an HTTP POST request.
    *
    * @param url     the URL.
    * @param body    the HTTP request body.
    * @param params  the request query parameters, if any.
    * @param headers the HTTP request headers, if any.
    *
    * @return the HTTP response body.
    */
   post(url: string, body?: any, params?: {[name: string]: string},
        headers?: {[name: string]: string}): Promise<R>;

   /**
    * Executes an HTTP PUT request.
    *
    * @param url     the URL.
    * @param body    the HTTP request body.
    * @param params  the request query parameters, if any.
    * @param headers the HTTP request headers, if any.
    *
    * @return the HTTP response body.
    */
   put(url: string, body?: any, params?: {[name: string]: string},
       headers?: {[name: string]: string}): Promise<R>;

   /**
    * Executes an HTTP GET request.
    *
    * @param url     the URL.
    * @param headers the HTTP req
    * @param params  the request query parameters, if any.uest headers, if any.
    *
    * @return the HTTP response body.
    */
   delete(url: string, params?: {[name: string]: string},
          headers?: {[name: string]: string}): Promise<R>;

   /**
    * Maps an HTTP response body onto a JSON type.
    *
    * @param response the HTTP response object.
    *
    * @return the HTTP response body.
    */
   map<T>(response: Promise<R>): Promise<T>;

   /**
    * Maps an HTTP response body onto plain text.
    *
    * @param response the HTTP response object.
    *
    * @return the HTTP response body.
    */
   mapText(response: Promise<R>): Promise<string>;

   /**
    * Maps an HTTP response body onto raw, binary data.
    *
    * @param response the HTTP response object.
    *
    * @return the HTTP response body.
    */
   mapBinary(response: Promise<R>): Promise<ArrayBuffer>;
}
