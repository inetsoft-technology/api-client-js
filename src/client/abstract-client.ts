/*
 * Copyright 2019 InetSoft Technology
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { HttpAdapter } from "./http-adapter";

/**
 * @hidden
 */
export abstract class AbstractClient {

   private get headers(): Promise<{[name: string]: string}> {
      return this.token().then((t) => ({ "X-Inetsoft-Api-Token": t }));
   }

   protected constructor(private adapter: HttpAdapter<any>, private token: () => Promise<string>,
                         private baseUrl: string) {}

   protected doGet<T>(url: string, params?: { [p: string]: string }): Promise<T> {
      return this.headers.then((h) =>
         this.adapter.map<T>(this.adapter.get(this.getUrl(url), params, h)));
   }

   protected doGetBinary(url: string, params?: { [p: string]: string }): Promise<ArrayBuffer> {
      return this.headers.then((h) =>
         this.adapter.mapBinary(this.adapter.get(this.getUrl(url), params, h)));
   }

   protected doPost<T>(url: string, data?: any, params?: { [p: string]: string }): Promise<T> {
      return this.headers.then((h) =>
         this.adapter.map<T>(this.adapter.post(this.getUrl(url), data, params, h)));
   }

   protected doPostBinary(url: string, data?: any,
                          params?: { [p: string]: string }): Promise<ArrayBuffer>
   {
      return this.headers.then((h) =>
         this.adapter.mapBinary(this.adapter.post(this.getUrl(url), data, params, h)));
   }

   protected doPut<T>(url: string, data?: any, params?: { [p: string]: string }): Promise<T> {
      return this.headers.then((h) =>
         this.adapter.map<T>(this.adapter.put(this.getUrl(url), data, params, h)));
   }

   protected doDelete<T>(url: string, params?: { [p: string]: string }): Promise<T> {
      return this.headers.then((h) =>
         this.adapter.map<T>(this.adapter.delete(this.getUrl(url), params, h)));
   }

   private getUrl(relativeUrl: string): string {
      return `${this.baseUrl}${relativeUrl}`;
   }
}
