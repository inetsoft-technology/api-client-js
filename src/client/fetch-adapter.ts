/*
 * Copyright 2019 InetSoft Technology
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { HttpAdapter } from "./http-adapter";

/**
 * @hidden
 */
export class FetchAdapter implements HttpAdapter<Response> {
   public get(url: string, params?: { [p: string]: string },
              headers?: { [p: string]: string }): Promise<Response>
   {
      const init = {method: "GET", headers};
      return fetch(this.appendParameters(url, params), init);
   }

   public post(url: string, body?: any, params?: { [p: string]: string },
               headers?: { [p: string]: string }): Promise<Response>
   {
      const init = {
         body: JSON.stringify(body),
         headers,
         method: "POST"
      };
      return fetch(this.appendParameters(url, params), init);
   }

   public put(url: string, body?: any, params?: { [p: string]: string },
              headers?: { [p: string]: string }): Promise<Response>
   {
      const init = {
         body: JSON.stringify(body),
         headers,
         method: "PUT"
      };
      return fetch(this.appendParameters(url, params), init);
   }

   public delete(url: string, params?: { [p: string]: string },
                 headers?: { [p: string]: string }): Promise<Response>
   {
      const init = {method: "DELETE", headers};
      return fetch(this.appendParameters(url, params), init);
   }

   public map<T>(response: Promise<Response>): Promise<T> {
      return response.then((r) => r.json());
   }

   public mapText(response: Promise<Response>): Promise<string> {
      return response.then((r) => r.text());
   }

   public mapBinary(response: Promise<Response>): Promise<ArrayBuffer> {
      return response.then((r) => r.arrayBuffer());
   }

   private appendParameters(url: string, params?: { [p: string]: string }): string {
      if (params) {
         const keys = Object.keys(params);

         if (keys.length) {
            let query = "?";

            for (let i = 0; i < keys.length; i++) {
               if (i > 0) {
                  query += "&";
               }

               query += encodeURIComponent(keys[i]) + "=" + encodeURIComponent(params[keys[i]]);
            }

            return url + query;
         }
      }

      return url;
   }
}
