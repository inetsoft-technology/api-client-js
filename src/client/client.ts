/*
 * Copyright 2019 InetSoft Technology
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { DatabaseApi } from "./database/database-api";
import { DatabaseClient } from "./database/database-client";
import { DataSourceApi } from "./datasource/data-source-api";
import { DataSourceClient } from "./datasource/data-source-client";
import { FetchAdapter } from "./fetch-adapter";
import { FileApi } from "./file/file-api";
import { FileClient } from "./file/file-client";
import { HttpAdapter } from "./http-adapter";
import { PrinterApi } from "./printer/printer-api";
import { PrinterClient } from "./printer/printer-client";
import { ReportApi } from "./report/report-api";
import { ReportClient } from "./report/report-client";
import { ScheduleApi } from "./schedule/schedule-api";
import { ScheduleClient } from "./schedule/schedule-client";
import { ServerApi } from "./server/server-api";
import { ServerClient } from "./server/server-client";
import { ViewsheetApi } from "./sheet/viewsheet/viewsheet-api";
import { ViewsheetClient } from "./sheet/viewsheet/viewsheet-client";
import { WorksheetApi } from "./sheet/worksheet/worksheet-api";
import { WorksheetClient } from "./sheet/worksheet/worksheet-client";

/**
 * @hidden
 */
class TokenProvider {
   private token: string | undefined;

   constructor(private serverUrl: string, private username: string | null,
               private password: string | null, private adapter: HttpAdapter<any>)
   {
   }

   public get(): Promise<string> {
      if(this.token) {
         return Promise.resolve(this.token);
      }

      const url = `${this.serverUrl}api/public/login`;
      let response: Promise<any> | undefined;

      if(this.username) {
         const data = {
            username: this.username,
            password: this.password
         };
         response = this.adapter.post(url, data);
      }
      else {
         response = this.adapter.get(url);
      }

      return this.adapter.map<{token: string}>(response).then(data => {
         this.token = data.token;
         return data.token;
      });
   }
}

/**
 * ApiClient is the client for the InetSoft Web APIs.
 */
export interface ApiClient {
   /**
    * The database API client.
    */
   database: DatabaseApi;

   /**
    * The data source API client.
    */
   dataSource: DataSourceApi;

   /**
    * The file API client.
    */
   file: FileApi;

   /**
    * The printer API client.
    */
   printer: PrinterApi;

   /**
    * The report API client.
    */
   report: ReportApi;

   /**
    * The schedule API client.
    */
   schedule: ScheduleApi;

   /**
    * The server API client.
    */
   server: ServerApi;

   /**
    * The viewsheet API client.
    */
   viewsheet: ViewsheetApi;

   /**
    * The worksheet API client.
    */
   worksheet: WorksheetApi;
}

/**
 * Implementation of the ApiClient interface.
 *
 * @hidden
 */
class ApiClientImpl implements ApiClient {

   public readonly database: DatabaseApi;
   public readonly dataSource: DataSourceApi;
   public readonly file: FileApi;
   public readonly printer: PrinterApi;
   public readonly report: ReportApi;
   public readonly schedule: ScheduleApi;
   public readonly server: ServerApi;
   public readonly viewsheet: ViewsheetApi;
   public readonly worksheet: WorksheetApi;

   constructor(serverUrl: string, username: string | null, password: string | null,
               adapter: HttpAdapter<any> | null)
   {
      adapter = adapter || new FetchAdapter();
      const provider = new TokenProvider(serverUrl, username, password, adapter);
      const token = () => provider.get();
      this.database = new DatabaseClient(adapter, token, serverUrl);
      this.dataSource = new DataSourceClient(adapter, token, serverUrl);
      this.file = new FileClient(adapter, token, serverUrl);
      this.printer = new PrinterClient(adapter, token, serverUrl);
      this.report = new ReportClient(adapter, token, serverUrl);
      this.schedule = new ScheduleClient(adapter, token, serverUrl);
      this.server = new ServerClient(adapter, token, serverUrl);
      this.viewsheet = new ViewsheetClient(adapter, token, serverUrl);
      this.worksheet = new WorksheetClient(adapter, token, serverUrl);
   }
}

/**
 * Creates a new instance of ApiClient.
 *
 * @param serverUrl the base URL of the InetSoft server.
 * @param username  the user name used to authenticate with the server. This parameter is not
 *                  required if anonymous access is allowed or if the user is already
 *                  authenticated in the enclosing web page.
 * @param password  the password used to authenticate with the server.
 * @param adapter   the adapter for the HTTP client. If not specified, this will use the fetch API.
 *
 * @return an new API client instance.
 */
export function createApiClient(serverUrl: string, username: string | null = null,
                                password: string | null = null,
                                adapter: HttpAdapter<any> | null = null): ApiClient {
   if(!serverUrl || !serverUrl.length) {
      throw new Error("The serverUrl parameter is required");
   }

   if(serverUrl[serverUrl.length - 1] !== "/") {
      serverUrl = serverUrl + "/";
   }

   return new ApiClientImpl(serverUrl, username, password, adapter);
}
