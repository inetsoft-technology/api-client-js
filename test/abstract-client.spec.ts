/*
 * Copyright 2019 InetSoft Technology
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { AbstractClient } from "../src/client/abstract-client";
import { HttpAdapter } from "../src/api-client";

class TestClient extends AbstractClient {

   constructor(adapter: HttpAdapter<any>, token: () => Promise<string>, baseUrl: string) {
      super(adapter, token, baseUrl);
   }

   get<T>(url: string, params?: { [p: string]: string }): Promise<T> {
      return this.doGet(url, params);
   }

   getBinary(url: string, params?: { [p: string]: string }): Promise<ArrayBuffer> {
      return this.doGetBinary(url, params);
   }

   post<T>(url: string, data?: any, params?: { [p: string]: string }): Promise<T> {
      return this.doPost(url, data, params);
   }

   postBinary(url: string, data?: any, params?: { [p: string]: string }): Promise<ArrayBuffer> {
      return this.doPostBinary(url, data, params);
   }

   put<T>(url: string, data?: any, params?: { [p: string]: string }): Promise<T> {
      return this.doPut(url, data, params);
   }

   delete<T>(url: string, params?: { [p: string]: string }): Promise<T> {
      return this.doDelete(url, params);
   }
}

describe("AbstractClient", () => {
   let adapter: any;
   let client: TestClient;

   beforeEach(() => {
      adapter = {
         get: jest.fn(),
         post: jest.fn(),
         put: jest.fn(),
         delete: jest.fn(),
         map: jest.fn(),
         mapText: jest.fn(),
         mapBinary: jest.fn()
      };
      client = new TestClient(adapter, () => Promise.resolve("secret"), "http://localhost:8080/sree/");
   });

   describe("when doGet() is called", () => {
      it("then the adapter is called with the correct arguments", () => {
         adapter.get.mockReturnValueOnce(Promise.resolve({result: "value"}));
         adapter.map.mockImplementation((promise: Promise<any>) => promise.then(response => ({mapped: response})));

         return client.get("test", { p1: "v1", p2: "v2" }).then(result => {
            expect(result).toMatchSnapshot();
            expect(adapter.get.mock.calls.length).toBe(1);
            expect(adapter.get.mock.calls[0][0]).toBe("http://localhost:8080/sree/test");
            expect(adapter.get.mock.calls[0][1]).toMatchSnapshot();
            expect(adapter.get.mock.calls[0][2]).toMatchSnapshot();
         });
      });
   });

   describe("when doGetBinary() is called", () => {
      it("then the adapter is called with the correct arguments", () => {
         adapter.get.mockReturnValueOnce(Promise.resolve({result: createBinaryData()}));
         adapter.mapBinary.mockImplementation((promise: Promise<any>) => promise.then(response => response.result));

         return client.getBinary("test", {p1: "v1", p2: "v2"}).then(result => {
            validateBinaryData(result);
            expect(adapter.get.mock.calls.length).toBe(1);
            expect(adapter.get.mock.calls[0][0]).toBe("http://localhost:8080/sree/test");
            expect(adapter.get.mock.calls[0][1]).toMatchSnapshot();
            expect(adapter.get.mock.calls[0][2]).toMatchSnapshot();
         });
      });
   });

   describe("when doPost() is called", () => {
      it("then the adapter is called with the correct arguments", () => {
         adapter.post.mockReturnValueOnce(Promise.resolve({result: "value" }));
         adapter.map.mockImplementation((promise: Promise<any>) => promise.then(response => ({mapped: response})));

         return client.post("test", { query: "value" }, { p1: "v1", p2: "v2" }).then(result => {
            expect(result).toMatchSnapshot();
            expect(adapter.post.mock.calls.length).toBe(1);
            expect(adapter.post.mock.calls[0][0]).toBe("http://localhost:8080/sree/test");
            expect(adapter.post.mock.calls[0][1]).toMatchSnapshot();
            expect(adapter.post.mock.calls[0][2]).toMatchSnapshot();
            expect(adapter.post.mock.calls[0][3]).toMatchSnapshot();
         });
      });
   });

   describe("when doPostBinary() is called", () => {
      it("then the adapter is called with the correct arguments", () => {
         adapter.post.mockReturnValueOnce(Promise.resolve({result: createBinaryData()}));
         adapter.mapBinary.mockImplementation((promise: Promise<any>) => promise.then(response => response.result));

         return client.postBinary("test", { query: "value" }, {p1: "v1", p2: "v2"}).then(result => {
            validateBinaryData(result);
            expect(adapter.post.mock.calls.length).toBe(1);
            expect(adapter.post.mock.calls[0][0]).toBe("http://localhost:8080/sree/test");
            expect(adapter.post.mock.calls[0][1]).toMatchSnapshot();
            expect(adapter.post.mock.calls[0][2]).toMatchSnapshot();
            expect(adapter.post.mock.calls[0][3]).toMatchSnapshot();
         });
      });
   });

   describe("when doPut() is called", () => {
      it("then the adapter is called with the correct arguments", () => {
         adapter.put.mockReturnValueOnce(Promise.resolve({result: "value" }));
         adapter.map.mockImplementation((promise: Promise<any>) => promise.then(response => ({mapped: response})));

         return client.put("test", { query: "value" }, { p1: "v1", p2: "v2" }).then(result => {
            expect(result).toMatchSnapshot();
            expect(adapter.put.mock.calls.length).toBe(1);
            expect(adapter.put.mock.calls[0][0]).toBe("http://localhost:8080/sree/test");
            expect(adapter.put.mock.calls[0][1]).toMatchSnapshot();
            expect(adapter.put.mock.calls[0][2]).toMatchSnapshot();
            expect(adapter.put.mock.calls[0][3]).toMatchSnapshot();
         });
      });
   });

   describe("when doDelete() is called", () => {
      it("then the adapter is called with the correct arguments", () => {
         adapter.delete.mockReturnValueOnce(Promise.resolve({result: "value"}));
         adapter.map.mockImplementation((promise: Promise<any>) => promise.then(response => ({mapped: response})));

         return client.delete("test", { p1: "v1", p2: "v2" }).then(result => {
            expect(result).toMatchSnapshot();
            expect(adapter.delete.mock.calls.length).toBe(1);
            expect(adapter.delete.mock.calls[0][0]).toBe("http://localhost:8080/sree/test");
            expect(adapter.delete.mock.calls[0][1]).toMatchSnapshot();
            expect(adapter.delete.mock.calls[0][2]).toMatchSnapshot();
         });
      });
   });
});

function createBinaryData(): ArrayBuffer {
   const buffer = new ArrayBuffer(16);
   const array = new Int32Array(buffer);
   array[0] = 1;
   array[1] = 2;
   array[2] = 3;
   array[3] = 4;
   return buffer;
}

function validateBinaryData(buffer: ArrayBuffer): void {
   expect(buffer).toBeTruthy();
   expect(buffer.byteLength).toBe(16);
   const actual = new Int32Array(buffer);
   expect(actual[0]).toBe(1);
   expect(actual[1]).toBe(2);
   expect(actual[2]).toBe(3);
   expect(actual[3]).toBe(4);
}
