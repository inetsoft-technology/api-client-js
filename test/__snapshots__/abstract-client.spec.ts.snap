// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbstractClient when doDelete() is called then the adapter is called with the correct arguments 1`] = `
Object {
  "mapped": Object {
    "result": "value",
  },
}
`;

exports[`AbstractClient when doDelete() is called then the adapter is called with the correct arguments 2`] = `
Object {
  "p1": "v1",
  "p2": "v2",
}
`;

exports[`AbstractClient when doDelete() is called then the adapter is called with the correct arguments 3`] = `
Object {
  "X-Inetsoft-Api-Token": "secret",
}
`;

exports[`AbstractClient when doGet() is called then the adapter is called with the correct arguments 1`] = `
Object {
  "mapped": Object {
    "result": "value",
  },
}
`;

exports[`AbstractClient when doGet() is called then the adapter is called with the correct arguments 2`] = `
Object {
  "p1": "v1",
  "p2": "v2",
}
`;

exports[`AbstractClient when doGet() is called then the adapter is called with the correct arguments 3`] = `
Object {
  "X-Inetsoft-Api-Token": "secret",
}
`;

exports[`AbstractClient when doGetBinary() is called then the adapter is called with the correct arguments 1`] = `
Object {
  "p1": "v1",
  "p2": "v2",
}
`;

exports[`AbstractClient when doGetBinary() is called then the adapter is called with the correct arguments 2`] = `
Object {
  "X-Inetsoft-Api-Token": "secret",
}
`;

exports[`AbstractClient when doPost() is called then the adapter is called with the correct arguments 1`] = `
Object {
  "mapped": Object {
    "result": "value",
  },
}
`;

exports[`AbstractClient when doPost() is called then the adapter is called with the correct arguments 2`] = `
Object {
  "query": "value",
}
`;

exports[`AbstractClient when doPost() is called then the adapter is called with the correct arguments 3`] = `
Object {
  "p1": "v1",
  "p2": "v2",
}
`;

exports[`AbstractClient when doPost() is called then the adapter is called with the correct arguments 4`] = `
Object {
  "X-Inetsoft-Api-Token": "secret",
}
`;

exports[`AbstractClient when doPostBinary() is called then the adapter is called with the correct arguments 1`] = `
Object {
  "query": "value",
}
`;

exports[`AbstractClient when doPostBinary() is called then the adapter is called with the correct arguments 2`] = `
Object {
  "p1": "v1",
  "p2": "v2",
}
`;

exports[`AbstractClient when doPostBinary() is called then the adapter is called with the correct arguments 3`] = `
Object {
  "X-Inetsoft-Api-Token": "secret",
}
`;

exports[`AbstractClient when doPut() is called then the adapter is called with the correct arguments 1`] = `
Object {
  "mapped": Object {
    "result": "value",
  },
}
`;

exports[`AbstractClient when doPut() is called then the adapter is called with the correct arguments 2`] = `
Object {
  "query": "value",
}
`;

exports[`AbstractClient when doPut() is called then the adapter is called with the correct arguments 3`] = `
Object {
  "p1": "v1",
  "p2": "v2",
}
`;

exports[`AbstractClient when doPut() is called then the adapter is called with the correct arguments 4`] = `
Object {
  "X-Inetsoft-Api-Token": "secret",
}
`;
